### ISSUE TITLE
point to the same tag as the requested api level in the url for "view source"

### ISSUE THREAD

**Issue by yrizk** [romannurik/AndroidSDKSearchExtension#28]
basically change /ref/heads/master to the right tag level found [here](https://android.googlesource.com/platform/frameworks/support/+refs) . Its something I find myself doing a quite a bit. Also, I think professional android developers will realize that its a good thing to compare implementation of a class across api levels. This would only apply to class and resource paths if the user decides he wants to look at google source code instead of git. Adding git would be trivial. 

Additions would be to... 
[ ] make /refs/heads/master a placeholder  and fill it in with the current api level. 
[ ] register a onChange listener for that api selector dropdown 
[ ] map that number to the latest tag and update the url if it changes. 

Things to consider: 
- new tags are added to AOSP as new rcs are made. The Android Team doesn't have a consistent rule or algorithm to determine when the the api level shifts to a new version of android. This means  someone will have to go in and update the map to point to the right tag in the url. Obviously keeping current behaviour in states of uncertainty is advised. 
@romannurik , I'm curious if to see if you have ideas on this. 

- What if that class is not in that level?
Previously, this was never a problem. nothing ever gets deleted, and head tracks all things. Now it will certainly break if you attempt to view JobService in api 12 or . At the same time, there is a big warning when you attempt to toggle into a lower api level, so I say the user was warned. Otherwise, I could write a function to determine if this class is unsupported and default to master in that case.

**Comment by romannurik** [comment]
I likely won't have time to work on this but very open to pull requests! Seems like there'll need to be some sort of mapping file between branches and API levels, no immediate ideas on automating that though. Re: classes missing, could possibly insert HTML content into 404 pages that offers suggested alternative links (e.g. "This class seems to be missing... try the master branch")

**Comment by yrizk** [comment]
Yes that sound good. To clarify, I was gauging interest in a pull request
if I wrote it. I started it already ( getting the value from the drop down
) ,but the mapping to release is unpredictable. I wish there was a formula
to calculate that conversion as opposed to hard coding it and constantly
updating.
